### PR TITLE
Fix errors in V7

### PR DESCRIPTION
**Pull request by EhsanKia**
- Don't try to run anon-chat for every single user connecting to chat
- Set eventlisteners max handlers to 100
- Check for undefined values in a few places

### CODE REVIEW

File: src/modules/anon_chat/index.js [hunk 1]
```diff
@@ -46,16 +46,17 @@ class AnonChatModule {
...
-        } else {
+        } else if (force) {
             this.join();
         }
     }
 
     onMessage($el, data) {
-        if (ignoreNextDC && data.message.includes('unable to connect to chat')) {
+        if (!ignoreNextDC) return;
+        if (data && data.message && data.message.includes('unable to connect to chat')) {
```
Comment (night) [inline src/modules/anon_chat/index.js]: this event only emits if data exists?

File: src/modules/anon_chat/index.js [hunk 2]
```diff
@@ -46,16 +46,17 @@ class AnonChatModule {
...
+    load(force = false) {
         if (forcedURL || settings.get('anonChat')) {
             this.part();
-        } else {
+        } else if (force) {
             this.join();
         }
     }
 
     onMessage($el, data) {
```
Comment (night) [inline src/modules/anon_chat/index.js]: `{message}` instead

File: src/modules/anon_chat/index.js [hunk 3]
```diff
@@ -17,15 +17,15 @@ class AnonChatModule {
         this.enabled = false;
         watcher.on('load.chat', () => this.load());
         watcher.on('chat.message', ($el, emberView) => this.onMessage($el, emberView));
```
Comment (night) [inline src/modules/anon_chat/index.js]: just fyi this is not an emberView it's the msgObject

**Comment by EhsanKia** [comment]
Fixed.

**Pull request merged**